### PR TITLE
ISSUE #3032

### LINKED ISSUE #3032: Verify ViewMetadata permission for the user before displaying the metadata for the table/database
**Feature**
#2907

**Describe the task**
Verify ViewMetadata permission for the user before displaying the metadata for the table/database

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes:

Fixes #3032

## Summary

  - Introduce **dynamic sampling** for the profiler, allowing sample percentages to adapt automatically based on table row count instead of requiring static configuration
  - Add `smartSampling` mode with built-in tiered thresholds (100% for ≤100K rows, down to 0.1% for >1B rows) and custom threshold-based mode for user-defined breakpoints
  - Add table metric computers for **Exasol, Teradata, Trino/Presto/Athena, Hive, Impala, and Databricks** to enable fast row count retrieval from system tables/stats commands
  - Refactor sampler architecture: `get_sample_query()` and `set_tablesample()` now receive resolved `StaticSamplingConfig` externally, eliminating repeated config resolution across subclasses
  - Replace flat `profileSample`/`profileSampleType`/`samplingMethodType` fields in `testSuitePipeline.json` with structured `profileSampleConfig` reference (backward compatible via `_resolve_profile_sample_config`)
 - Set smart sampling as default config for profiler
 - Added smart sampling config in advance config

https://github.com/user-attachments/assets/7153a844-3569-4d4a-9568-409e8244ff24


  ## Details

  ### Dynamic Sampling Configuration

  The new `dynamicSamplingConfig.json` schema supports two modes:

  1. **Smart Sampling** (`smartSampling: true`): Automatically selects sampling percentage based on row count tiers
  2. **Custom Thresholds** (`smartSampling: false`): User defines `rowCountThreshold → profileSample` mappings; highest applicable threshold wins. Configuration hierarchy remains: entity config → table profiler config → schema config → database config → pipeline default.               
                  
  ### Row Count Fetching for Dynamic Resolution                                                                                              
                  
  `SQASampler._get_asset_row_count()` leverages `table_metric_computer_factory` to dispatch to dialect-specific system tables (pg_class, sys.partitions, SHOW STATS, etc.) for fast row counts without full COUNT(*) scans. Falls back to COUNT(*) for unregistered dialects or partitioned tables.                                                                                                                        
                  
  ### Sampler Refactoring

  - `_resolve_sample_config` cached property on `SamplerInterface` centralizes dynamic→static resolution                                     
  - Subclass samplers (MSSQL, BigQuery, Snowflake, Postgres, AzureSQL) receive resolved config via parameters instead of re-reading from `self.sample_config`                                                                                                                       
  - BurstIQ sampler refactored to extend `DatalakeSampler`, eliminating ~60 lines of duplicated logic
                                                                                                                                             
  ## Test plan                                                                                                                               
                                                                                                                                             
  - [x] Unit tests for `resolve_static_sampling_config` — all tier boundaries, threshold matching, edge cases (62 tests)                     
  - [x] Unit tests for new table metric computers — Exasol, Teradata, Trino, Hive, Impala, Databricks
  - [x] Unit tests for all database-specific samplers (MSSQL, AzureSQL, BigQuery, Postgres, Snowflake)                                       
  - [x] Integration tests against real PostgreSQL — row count accuracy via metric computer + dynamic resolution                              
  - [x] Integration tests against real MSSQL — metric computer + TABLESAMPLE execution                                                       
  - [x] Integration tests against real ClickHouse, TimescaleDB, Hive — metric computer accuracy                                                                            
  - [ ] Verify backward compatibility: existing static configs continue working unchanged
  - [x] Manual verification of Snowflake, BigQuery

#
### Type of change:
<!-- You should choose 1 option and delete options that aren't relevant -->
- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:
<!-- add an x in [] if done, don't mark items that you didn't do !-->
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

<!-- Based on the type(s) of your change, uncomment the required checklist 👇 -->

<!-- Bug fix
- [ ] I have added a test that covers the exact scenario we are fixing. For complex issues, comment the issue number in the test for future reference.
-->

Improvement
- [x] I have added tests around the new logic.
- [x] For connector/ingestion changes: I updated the documentation.

<!-- New feature
- [ ] The issue properly describes why the new feature is needed, what's the goal, and how we are building it. Any discussion
    or decision-making process is reflected in the issue.
- [ ] I have updated the documentation.
- [ ] I have added tests around the new logic.
-->

<!-- Breaking change
- [ ] I have added the tag `Backward-Incompatible-Change`.
-->

----
## Summary by Gitar

- **Internationalization:**
  - Translated `smart-sampling` and `smart-sampling-hint` i18n keys across 18 localized language files.
  - Added `cache-service-not-configured-message` warning to locale files.
- **UI/UX:**
  - Updated `ProfileSampleConfigField.test.tsx` to reflect smart sampling as the default configuration.
  - Added mandatory `smartSampling` boolean field to dynamic configuration tests and test mocks.

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json [hunk 1]
```diff
@@ -45,19 +45,8 @@
...
-    "profileSampleType": {
-      "$ref": "../type/basic.json#/definitions/profileSampleType",
-      "title": "Profile Sample Type"
-    },
-    "samplingMethodType": {
-      "$ref": "../type/basic.json#/definitions/samplingMethodType",
-      "title": "Sampling Method Type"
+    "profileSampleConfig": {
+      "$ref": "../type/samplingConfig.json#/definitions/profileSampleConfig"
     },
```
[the remark this answers is not in the document]
Comment (TeddyCr) [reply openmetadata-spec/src/main/resources/json/schema/metadataIngestion/testSuitePipeline.json:50]: Handled in migration steps

**Review by pmbrull — COMMENTED** [review]
lgtm, just one q, if we set the dynamic sampling will we be able to see that in the profile page of a table / as part of sampling results? to differentiate what's manually added from a user vs. set by the system as a result

**Review by pmbrull — DISMISSED** [review]

**Comment by TeddyCr** [comment]
> lgtm, just one q, if we set the dynamic sampling will we be able to see that in the profile page of a table / as part of sampling results? to differentiate what's manually added from a user vs. set by the system as a result

Not yet, but that would be a good improvement

File: openmetadata-ui/src/main/resources/ui/src/locale/languages/fr-fr.json [hunk 2]
```diff
@@ -1970,6 +1970,7 @@
     "slack-support": "Support Slack",
     "slash-symbol": "/",
     "small": "Petit",
+    "smart-sampling": "Smart Sampling",
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/locale/languages/fr-fr.json]: can you please translate new key's value to appropriate language?

**Review by pmbrull — DISMISSED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]
